src/main/scala/loamstream/drm/uger/QacctAccountingClient.scala

```diff
@@ -63,31 +107,26 @@ object QacctAccountingClient extends Loggable {
   /**
    * Make a UgerAcct that will retrieve job metadata by running some executable, by default, `qacct`.
    */
-  def useActualBinary(binaryName: String = "qacct"): QacctAccountingClient = {
+  def useActualBinary(ugerConfig: UgerConfig, binaryName: String = "qacct"): QacctAccountingClient = {
     import scala.sys.process._
 
-    def getQacctOutputFor(jobId: String): Seq[String] = {
+    def invokeQacctFor(jobId: String): Seq[String] = {
       val tokens = makeTokens(binaryName, jobId)
 
       val noopProcessLogger = ProcessLogger(_ => ())
 
-      try {
-        //NB: Use noopProcessLogger to suppress stderr output that would otherwise go to the console.
-        //NB: Even with noopProcessLogger, all lines written to stdout will be available via lineStream.
-        //NB: Use toIndexedSeq to eagerly consume stdout, effectively running the command synchronously.
-        tokens.lineStream(noopProcessLogger).toIndexedSeq
-      } catch {
-        case NonFatal(e) => {
-          debug(s"Error invoking '$binaryName'; execution node and queue won't be available: $e")
-          trace(s"'$binaryName' invocation failure stack trace:", e)
-
-          Seq.empty
-        }
-      }
+      //NB: Use noopProcessLogger to suppress stderr output that would otherwise go to the console.
+      //NB: Even with noopProcessLogger, all lines written to stdout will be available via lineStream.
+      //NB: Use toIndexedSeq to eagerly consume stdout, effectively running the command synchronously.
+      tokens.lineStream(noopProcessLogger).toIndexedSeq
     }
-
-    new QacctAccountingClient(getQacctOutputFor)
+    
+    new QacctAccountingClient(ugerConfig, invokeQacctFor)
   }
 
   private[uger] def makeTokens(binaryName: String, jobId: String): Seq[String] = Seq(binaryName, "-j", jobId)
+  
+  private[uger] def delaySequence: Iterator[Duration] = {
+    Iterator.iterate(0.5)(_ * 2).map(scala.math.min(30, _)).map(_.seconds)
```

Comment:
Not that it matters, but you don't need all the maps as time durations handle the math just fine:
```
@ Iterator.iterate(0.5.seconds)(d => (d*2).min(30.seconds)).take(8).to[List]
res9: List[FiniteDuration] = List(
  500 milliseconds,
  1000 milliseconds,
  2000 milliseconds,
  4000 milliseconds,
  8000 milliseconds,
  16000 milliseconds,
  30 seconds,
  30 seconds
)
```

Reply:
Aha, with `Duration`s throughout.  Your way is more declarative, I'll use that.